xreds/utils.py

```diff
@@ -1,74 +1,112 @@
+        return "zarr"
 
-    return 'unknown'
+    return "unknown"
 
 
-def load_dataset(dataset_spec: dict) -> xr.Dataset | None:
+def load_dataset(
+    dataset_spec: dict, redis_cache: Optional[Redis] = None, cache_timeout: int = 600
+) -> xr.Dataset | None:
     """Load a dataset from a path"""
     ds = None
-    dataset_path = dataset_spec['path']
+    dataset_path = dataset_spec["path"]
     dataset_type = dataset_spec.get("type", None)
     if not dataset_type:
         dataset_type = infer_dataset_type(dataset_path)
         logger.info(f"Inferred dataset type {dataset_type} for {dataset_path}")
-    if dataset_type == 'unknown':
+    if dataset_type == "unknown":
         logger.error(f"Could not infer dataset type for {dataset_path}")
         return None
 
-    chunks = dataset_spec.get('chunks', None)
-    drop_variables = dataset_spec.get('drop_variables', None)
-    additional_coords = dataset_spec.get('additional_coords', None)
-    additional_attrs = dataset_spec.get('additional_attrs', None)
-    key = dataset_spec.get('key', None)
-    secret = dataset_spec.get('secret', None)
+    chunks = dataset_spec.get("chunks", None)
+    drop_variables = dataset_spec.get("drop_variables", None)
+    additional_coords = dataset_spec.get("additional_coords", None)
+    additional_attrs = dataset_spec.get("additional_attrs", None)
 
-    if dataset_type == 'netcdf':
-        ds = xr.open_dataset(dataset_path, engine='netcdf4', chunks=chunks, drop_variables=drop_variables)
+    if dataset_type == "netcdf":
+        engine = dataset_spec.get("engine", "netcdf4")
+        ds = xr.open_dataset(
+            dataset_path, engine=engine, chunks=chunks, drop_variables=drop_variables
+        )
         if additional_coords is not None:
             ds = ds.set_coords(additional_coords)
-    elif dataset_type == 'grib2':
-        ds = xr.open_dataset(dataset_path, engine='cfgrib')
-    elif dataset_type == 'kerchunk':
-        if key is not None:
-            options = {'anon': False, 'key': key, 'secret': secret}
+    elif dataset_type == "grib2":
+        # TODO: Network support?
+        ds = xr.open_dataset(dataset_path, engine="cfgrib")
+    elif dataset_type == "kerchunk":
+        target_protocol = dataset_spec.get("target_protocol", 's3')
+        target_options = dataset_spec.get("target_options", {'anon': True})
+        remote_protocol = dataset_spec.get("remote_protocol", 's3')
+        remote_options = dataset_spec.get("remote_options", {'anon': True})
+
+        if redis_cache is not None:
+            reference_url = f"rediscache::{dataset_path}"
+            with fsspec.open(
+                reference_url,
+                mode="rb",
+                rediscache={"redis": redis_cache, "expiry": cache_timeout},
+                s3=target_options,
+            ) as f:
+                refs = ujson.load(f)
+            fs = RedisCachingReferenceFileSystem(
+                redis=redis_cache,
+                expiry_time=180,
```

Comment:
Do we maybe want to make this configurable? `REDIS_CACHE_EXPIRY_TIME`?

Reply:
Yup good catch, that should use the expiry config for sure

Reply:
for now it will share the same value as `DATASET_CACHE_TIMEOUT`, unless you think the redis timeout should be different?